### ISSUE TITLE
Villages generate too many buildings from other mods

### ISSUE THREAD

**Issue by DPswxz** [jtorleon-studios-team/bettervillages#22]
I want the village to generate one or two same kinds of building in other mods, such as Bountiful's bounty boards , houses of Farmer's Delight, Four types of buildings in Better Villages, but in fact they generate a lot, even more than 6, which is unreasonable, causing the village It's too big.
![picture](https://user-images.githubusercontent.com/56027030/230707707-0f0049da-be73-4884-83ce-41ff310842e3.png)
![picture2](https://user-images.githubusercontent.com/56027030/230707708-a3f4dbb8-bd3f-46ba-a02c-c1a0336a8a1b.png)

**Comment by jtorleonstudios** [comment]
Yes we are working on compatibility mods that should arrive in the next few weeks. The next one to come out is bountifull.
While waiting for the compatibility, you can decrease the weight of houses in each mod concerned (in configuration file).

**Be sure to follow the updates to receive notifications when the compatibility is released**

Thanks you for your patience

**Comment by YumatanGames** [comment]
Yes, same issue here too. 
![Buggy Bounty](https://user-images.githubusercontent.com/80188726/232204046-9966dc8f-cb4b-4915-af25-84765f2d5014.jpg)

**Comment by jtorleonstudios** [comment]
Compat bountiful is under review: https://www.curseforge.com/minecraft/texture-packs/better-villages-bountiful

![image](https://user-images.githubusercontent.com/89711083/232342683-02bd4195-73a1-45a5-a654-eb60b242df95.png)

**Comment by jtorleonstudios** [comment]
Compat More Villages is under review: https://legacy.curseforge.com/minecraft/texture-packs/better-villages-more-villagers-compatibility

**Comment by LFMaozi** [comment]
Also for mods 
[More Villagers] https://www.curseforge.com/minecraft/mc-mods/more-villagers
[VillagersPlus ] https://www.curseforge.com/minecraft/mc-mods/villagersplus-forge
[Domestication Innovation] https://www.curseforge.com/minecraft/mc-mods/domestication-innovation
It seems all small structures from different mods share the same generation amount pool, which makes adjust their ratiol annoying. Maybe seperating the their generation could relieve this

**Comment by jtorleonstudios** [comment]
For More Villagers
> Compat More Villages is under review: https://legacy.curseforge.com/minecraft/texture-packs/better-villages-more-villagers-compatibility

Villager Plus is in progress

**Comment by polyjitter** [comment]
Something that could help is just giving us the option to dial the village size back down to the vanilla size, since these structure frequencies are built around that limit. Would have compatibility with most important mods.

**Comment by jtorleonstudios** [comment]
> Something that could help is just giving us the option to dial the village size back down to the vanilla size, since these structure frequencies are built around that limit. Would have compatibility with most important mods.

No generation parameters have been changed, so the size has not been changed. The problem is that our houses aren't the same size as the houses added by other mods, so other smaller houses respond more easily to the generation condition than ours. In any case, our choice is to make compatibility datapacks so that the houses adopt the same general size and architectural style.

**Comment by ZsoltMolnarrr** [comment]
I am fairly certain that this issue needs to be fixed by other developers, by making the structure weight of their new houses configurable.

I believe a correct implementation is the following:
https://github.com/ZsoltMolnarrr/Gazebo/blob/94380db004420db8f29f20bad4c55ab0e86c6b0c/common/src/main/java/net/gazebo/worldgen/GazeboWorldGen.java#L22

**Comment by raffle2** [comment]
Would also like to include https://www.curseforge.com/minecraft/mc-mods/chefs-delight-forge in your radar, I tried tweaking it as low as I could but it still spawns a bunch in villages.

**Comment by vinnyg0621** [comment]
there are too many major bugs in this mod... I **so** wish you would work on the time to work on them @jtl-elisa 😢

**Comment by tmslpm** [comment]
This issue is now closed:
- We have updated the existing compatibility packs.
- Compatibility packs that don't exist yet have been registered as separate issues and will be available in the future (follow the progress of the relevant issues).
- As a reminder, if the concerned mods provide a configuration, please check it to see if it’s possible to adjust the size of each house.
- Also, the latest update (3.3.0) slightly reduced the overall village size, but it’s still not enough to fully address the issue.

> I am fairly certain that this issue needs to be fixed by other developers, by making the structure weight of their new houses configurable. 
That’s true, but Better Villages also needs to be adapted accordingly.
